server/src/test/java/io/spine/client/CommandRequestTest.java

```diff
@@ -98,4 +110,105 @@ void rejections() {
+        }
+    }
+
+    @Nested
+    @DisplayName("Allow setting custom consumer error handler")
+    class CustomConsumerErrorHandler {
+
+        private boolean handlerInvoked;
+        private @Nullable Throwable passedThrowable;
+
+        @BeforeEach
+        void setup() {
+            handlerInvoked = false;
+            passedThrowable = null;
+        }
+
+        @Test
+        @DisplayName("rejecting `null`")
+        void rejectingNull() {
+            assertThrows(NullPointerException.class, () -> commandRequest.onConsumingError(null));
+        }
+
+        @Test
+        @DisplayName("invoking the handler when a consumer fails")
+        void invocation() {
+            ConsumerErrorHandler<EventMessage> handler = (c, th) -> {
+                handlerInvoked = true;
+                passedThrowable = th;
+            };
+            RuntimeException exception = new RuntimeException("Consumer-generated error.");
+
+            commandRequest.onConsumingError(handler)
+                          .observe(UserLoggedIn.class, e -> { throw exception; })
+                          .post();
+
+            assertThat(handlerInvoked)
+                    .isTrue();
+            assertThat(passedThrowable)
+                    .isEqualTo(exception);
+        }
+    }
+
+    @Nested
+    @DisplayName("Allow setting custom posting error handler")
+    class CustomPostingErrorHandler {
+
+        private @Nullable Message postedMessage;
+        private @Nullable Error returnedError;
+
+        @BeforeEach
+        void setup() {
+            postedMessage = null;
+            returnedError = null;
+        }
+
+        @Test
+        @DisplayName("rejecting `null`")
+        void rejectingNull() {
+            assertThrows(NullPointerException.class, () -> commandRequest.onPostingError(null));
+        }
+
+        @Test
+        @DisplayName("invoking handler when an invalid command posted")
+        void invocation() {
+            PostingErrorHandler handler = (message, error) -> {
+                postedMessage = message;
+                returnedError = error;
+            };
+
+            UnsupportedCommand commandMessage = UnsupportedCommand
+                    .newBuilder()
+                    .setUser(GivenUserId.generated())
+                    .build();
+            CommandRequest request =
+                    client().asGuest()
+                            .command(commandMessage)
+                            .onPostingError(handler);
+            request.post();
+
+            assertThat(returnedError)
+                    .isNotNull();
+            assertThat(postedMessage)
+                    .isInstanceOf(Command.class);
+            Command expected = Command
+                    .newBuilder()
+                    .setMessage(AnyPacker.pack(commandMessage))
+                    .build();
+            ProtoTruth.assertThat(postedMessage)
```

Comment:
I think, you can use the `static import` here as well. It should get resolved among similar methods.